src/Microsoft.AspNet.Hosting/WebHostBuilder.cs

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNet.Builder;
+using Microsoft.AspNet.Hosting.Builder;
+using Microsoft.AspNet.Hosting.Internal;
+using Microsoft.AspNet.Hosting.Server;
+using Microsoft.AspNet.Hosting.Startup;
+using Microsoft.Framework.ConfigurationModel;
+using Microsoft.Framework.DependencyInjection;
+using Microsoft.Framework.Logging;
+using Microsoft.Framework.Runtime;
+using Microsoft.Framework.Runtime.Infrastructure;
+
+namespace Microsoft.AspNet.Hosting
+{
+    public class WebHostBuilder
+    {
+        public const string EnvironmentKey = "ASPNET_ENV";
+
+        private readonly IServiceProvider _fallbackServices;
+        private readonly IHostingEnvironment _hostingEnvironment;
+        private readonly ILoggerFactory _loggerFactory;
+
+        private Action<IServiceCollection> _configureServices;
+        private IConfiguration _config;
+
+        // Only one of these should be set
+        private StartupMethods _startup;
+        private Type _startupType;
+        private string _startupAssemblyName;
+
+        // Only one of these should be set
+        private string _serverFactoryLocation;
+        private IServerFactory _serverFactory;
+
+        public WebHostBuilder() : this(null) { }
+
+        public WebHostBuilder(IServiceProvider fallbackServices)
+        {
+            _hostingEnvironment = new HostingEnvironment();
+            _loggerFactory = new LoggerFactory();
+            _fallbackServices = fallbackServices ?? CallContextServiceLocator.Locator.ServiceProvider;
```

Comment:
Drop support for CallContextServiceLocator.Locator.ServiceProvider, we're removing this across the board. It can stay in test host for now but all mainline APIs should remove it.